src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/FootnoteServiceTests.cs

```diff
@@ -210,450 +198,304 @@ await using (var statisticsDbContext = InMemoryStatisticsDbContext(contextId))
         [Fact]
         public async Task CreateFootnote_MultipleFootnotesHaveExpectedOrder()
         {
-            var subject = new Subject();
+            var release = _releaseGenerator.Generate();
+            var subject = _subjectsGenerator.Generate();
 
-            // Create a release which already has some existing footnotes
-            var release = new Release
-            {
-                Footnotes = new List<ReleaseFootnote>
-                {
-                    new()
-                    {
-                        Footnote = new Footnote
-                        {
-                            Content = "Footnote 1",
-                            Order = 0,
-                            Subjects = new List<SubjectFootnote>
-                            {
-                                new()
-                                {
-                                    Subject = subject
-                                }
-                            }
-                        }
-                    },
-                    new()
-                    {
-                        Footnote = new Footnote
-                        {
-                            Content = "Footnote 2",
-                            Order = 1,
-                            Subjects = new List<SubjectFootnote>
-                            {
-                                new()
-                                {
-                                    Subject = subject
-                                }
-                            }
-                        }
-                    }
-                }
-            };
-
-            var releaseSubject = new ReleaseSubject
-            {
-                Release = release,
-                Subject = subject
-            };
+            var footnotes = _footnoteGenerator
+                .WithSubjects(ListOf(subject))
+                .GenerateList(2);
 
-            var contextId = Guid.NewGuid().ToString();
+            var releaseSubject = _releaseSubjectGenerator
+                .ForInstance(s => s.Set(rs => rs.Release, release))
+                .ForInstance(s => s.Set(rs => rs.Subject, subject))
+                .Generate();
 
+            // Create a release which already has some existing footnotes
+            var releaseFootnotes = _releaseFootnoteGenerator
+                .ForInstance(s => s.Set(rf => rf.Release, release))
+                .ForInstance(s => s.Set(rf => rf.Footnote, (_, _, context) => footnotes[context.Index]))
+                .GenerateList(footnotes.Count);
```

Comment:
See my other response above.  In this case it's not enough to be able to do this suggestion above because I need a ReleaseFootnote *per* Footnote that has already been generated in the pre-existing `footnotes` List.

Although it looks like potentially you're maybe showing one of the suggestions here, which is having an extension method that takes a List of Footnotes and distributes them across the generated ReleaseFootnotes using the SetterContext.Index?  In my example I would've called this method "WithFootnotes".

Reply:
(as per my other response above) OK so you're saying try to eliminate the need to have a pre-existing `footnotes` List in the first place by creating them all in situ here.

Yeah I'm happy to do that - there may be cases where we end up with some pretty deep nesting of data generation if we always stick to this pattern, but I'm happy to give it a go and see what it looks like.

Reply:
Hey @ntsim - see my answer here, thanks: https://github.com/dfe-analytical-services/explore-education-statistics/pull/3889#discussion_r1112101438

Reply:
Done ta - see reply to the link above